使用pnpm运行报错

<!-- 请不要删除自动生成的 Issue 标签 -->
<!-- 请不要删除自动生成的 Issue 标签 -->
### 相关平台
H5

### 复现仓库
[https://github.com/PL-FE/taro-h5.git](https://github.com/PL-FE/taro-h5.git)
**浏览器版本:  Chrome111.0.5563.147**
**使用框架: React**

### 复现步骤
1、pnpm install
2 、pnpm run dev:h5

### 期望结果
使用pnpm install 后，可以顺利运行

### 实际结果
使用yarn install 后，运行不报错
使用pnpm install后，运行报错
   Failed to compile.

./src/app.config.js
Module not found: Can't resolve '@tarojs/taro' in 'D:\csdn\taro-h5\src'

### 环境信息
```
 Taro CLI 3.4.14 environment info:
    System:
      OS: Windows 10
    Binaries:
      Node: 16.13.1 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.19 - C:\Users\pengl\AppData\Local\pnpm\yarn.CMD
      npm: 8.1.2 - C:\Program Files\nodejs\npm.CMD
```

<!-- generated by taro-issues. 请勿修改或删除此行注释 --><!--labels=T-h5,V-3,F-react-->

--- a/packages/taro-components-advanced/rollup.config.js
+++ b/packages/taro-components-advanced/rollup.config.js
@@ -22,7 +22,9 @@ export default {
   treeshake: false,
   plugins: [
     externals({
-      devDeps: false
+      deps: true,
+      devDeps: false,
+      include: ['react', 'react-dom', 'vue']
     }),
     resolve({
       preferBuiltins: false,

--- a/packages/taro-components-library-react/rollup.config.js
+++ b/packages/taro-components-library-react/rollup.config.js
@@ -16,7 +16,8 @@ const config = {
   treeshake: false,
   plugins: [
     externals({
-      devDeps: false
+      devDeps: false,
+      include: /^react/
     }),
     resolve({
       preferBuiltins: false,

--- a/packages/taro-components-library-react/rollup.config.js
+++ b/packages/taro-components-library-react/rollup.config.js
@@ -16,8 +16,9 @@ const config = {
   treeshake: false,
   plugins: [
     externals({
+      deps: true,
       devDeps: false,
-      include: /^react/
+      include: ['react', 'react-dom']
     }),
     resolve({
       preferBuiltins: false,

--- a/packages/taro-components-library-vue2/rollup.config.js
+++ b/packages/taro-components-library-vue2/rollup.config.js
@@ -16,8 +16,10 @@ const config = {
   treeshake: false,
   plugins: [
     externals({
+      deps: true,
       devDeps: false,
-      exclude: ['vue-fragment']
+      exclude: 'vue-fragment',
+      include: 'vue'
     }),
     resolve({
       preferBuiltins: false,

--- a/packages/taro-components-library-vue3/rollup.config.js
+++ b/packages/taro-components-library-vue3/rollup.config.js
@@ -16,7 +16,9 @@ const config = {
   treeshake: false,
   plugins: [
     externals({
-      devDeps: false
+      deps: true,
+      devDeps: false,
+      include: 'vue'
     }),
     resolve({
       preferBuiltins: false,

--- a/packages/taro-components-react/rollup.config.js
+++ b/packages/taro-components-react/rollup.config.js
@@ -10,6 +10,7 @@ export default {
   input: 'src/index.ts',
   plugins: [
     externals({
+      deps: true,
       devDeps: false,
       include: [/^react$/]
     }),

--- a/packages/taro-loader/src/h5.ts
+++ b/packages/taro-loader/src/h5.ts
@@ -30,7 +30,7 @@ export default function (this: webpack.LoaderContext<any>) {
 
   const pathDirname = dirname(this.resourcePath)
   const pageName = isMultiRouterMode ? join(pathDirname, options.filename).replace(options.sourceDir + '/', '') : ''
-  const pages: Map<string, string> = options.pages
+  const pages: Map<string, string> = new Map(options.pages)
   const pxTransformConfig = options.pxTransformConfig
   const runtimePath = Array.isArray(options.runtimePath) ? options.runtimePath : [options.runtimePath]
   let setReconcilerPost = ''

--- a/packages/taro-platform-h5/rollup.config.js
+++ b/packages/taro-platform-h5/rollup.config.js
@@ -40,6 +40,7 @@ const variesConfig = [{
     format: 'cjs'
   },
   plugins: getPlugins([externals({
+    deps: true,
     devDeps: false
   })])
 }, {
@@ -54,6 +55,7 @@ const variesConfig = [{
     preserveModulesRoot: 'src'
   },
   plugins: getPlugins([externals({
+    deps: true,
     devDeps: false
   })])
 }, {
@@ -65,6 +67,8 @@ const variesConfig = [{
   },
   plugins: getPlugins([
     externals({
+      deps: true,
+      devDeps: false,
       exclude: ['@tarojs/components', '@tarojs/taro-h5']
     })
   ], [exportNameOnly()])

--- a/packages/taro-plugin-html/rollup.config.js
+++ b/packages/taro-plugin-html/rollup.config.js
@@ -6,7 +6,7 @@ const cwd = __dirname
 
 const base = {
   plugins: [
-    externals({ devDeps: false }),
+    externals({ deps: true, devDeps: false }),
     ts(),
   ]
 }

--- a/packages/taro-plugin-http/rollup.config.js
+++ b/packages/taro-plugin-http/rollup.config.js
@@ -16,6 +16,7 @@ const compileConfig = {
   },
   plugins: [
     externals({
+      deps: true,
       devDeps: false
     }),
     nodeResolve(),

